gadfly/sentence_summarizer.py

```diff
@@ -58,3 +59,92 @@ def summarize(self, sents, n):
     def _rank(self, ranking, n):
         """ return the first n sentences with highest ranking """
         return nlargest(n, ranking, key=ranking.get)
+
+
+class TF_IDFSummarizer:
+    def __init__(self):
+        pass
+
+    def get_tokens_and_freqs(self, sents, style):
+        """Get a list of all tokens in sents and a Counter of them"""

```

Comment:
You are intending the style parameter to be an option that the caller has to set to choose. However, in order for the caller to set this option, they have inspect this method to know what the acceptable values of this parameter are.

An alternative, better design, would be to use an Enum. You can create an Enum within this file that has these options stored as variables. For example:

```
from enum import Enum


class TokenStyle(Enum):
    standard = "standard"
    lower = "lower"
    lemma = "lemma"
```

Now, the caller can import this enum in the code they are using to ensure that the value that they are passing is the same value that your method is checking for, without having to inspect  your implementation. See how the GapFillBlankType enum is implemented and used in gap fill generator on development.

Here is more info on using enums (it's in Java but the point still stands): http://crunchify.com/why-and-for-what-should-i-use-enum-java-enum-examples/.